component/src/shielded_pool/component.rs

```diff
@@ -401,6 +474,115 @@ impl ShieldedPool {
             .await
             .expect("block height must be set")
     }
+
+    /// Get the current epoch.
+    async fn epoch(&self) -> Epoch {
+        // Get the height
+        let height = self.height().await;
+
+        // Get the epoch duration
+        let epoch_duration = self.epoch_duration().await;
+
+        // The current epoch
+        Epoch::from_height(height, epoch_duration)
+    }
+
+    /// Get the epoch duration.
+    async fn epoch_duration(&self) -> u64 {
+        self.state
+            .get_epoch_duration()
+            .await
+            .expect("can get epoch duration")
+    }
```

Comment:
The epoch isn't a property of the shielded pool, so it shouldn't be in the shielded pool's `View` trait. A better place would be the common `penumbra_chain::View` trait.